File: day5-handson/python/src/logger.py
# logger.py
#
# A simple application logger.
# In tests we SPY on this using patch(wraps=) to verify
# warning calls without replacing its real behaviour.


def info(message: str) -> None:
    """Logs an informational message."""
    print(f"[INFO]  {message}")


def warn(message: str) -> None:
    """Logs a warning message."""
    print(f"[WARN]  {message}")


def error(message: str) -> None:
    """Logs an error message."""
    print(f"[ERROR] {message}")

File: day5-handson/js/src/logger.js
/**
 * logger.js
 *
 * A simple application logger.
 * In tests we SPY on this to verify warning calls
 * without replacing its real behaviour.
 */

const logger = {
  /**
   * Logs an informational message.
   * @param {string} message
   */
  info(message) {
    console.log(`[INFO]  ${message}`);
  },

  /**
   * Logs a warning message.
   * @param {string} message
   */
  warn(message) {
    console.warn(`[WARN]  ${message}`);
  },

  /**
   * Logs an error message.
   * @param {string} message
   */
  error(message) {
    console.error(`[ERROR] ${message}`);
  },
};

module.exports = logger;
